linux/keyman-system-service/src/KeyboardDevice.cpp

```diff
@@ -0,0 +1,85 @@
+#include <dirent.h>
+#include <fcntl.h>
+#include <iostream>
+#include <string.h>
+#include <string>
+#include <unistd.h>
+#include "KeyboardDevice.h"
+
+using namespace std;
+
+KeyboardDevice::KeyboardDevice()
+{
+  this->dev = nullptr;
+  this->fd  = -1;
+  this->hasCapsLockLed = -1;
+  this->debug          = false;
+}
+
+KeyboardDevice::~KeyboardDevice()
+{
+  libevdev_free(this->dev);
```

Comment:
Is this safe to call in failure cases (e.g. `Initialize` fails and `this->dev` is never set)?